macos/Sources/Features/Terminal/TerminalController.swift

```diff
@@ -812,32 +812,37 @@ class TerminalController: BaseTerminalController, TabGroupCloseCoordinator.Contr
 
     /// Close all windows, asking for confirmation if necessary.
     static func closeAllWindows() {
-        let needsConfirm: Bool = all.contains {
-            $0.surfaceTree.contains { $0.needsConfirmQuit }
-        }
-
-        if (!needsConfirm) {
-            closeAllWindowsImmediately()
-            return
+        var confirmWindow: NSWindow?
+        var needsConfirm = false
+        for controller in all {
+            if let surfaceToConfirm = controller.surfaceTree.first(where: { $0.needsConfirmQuit }) {
+                needsConfirm = true
+                confirmWindow = surfaceToConfirm.window
+                break
```

Comment:
This can be changed into a functional call `all.first` and assigned to confirmWindow in a let to avoid mutable vars.